pkg/helm/metrics/metrics.go

```diff
@@ -0,0 +1,83 @@
+package metrics
+
+import (
+	"log"
+	"os"
+
+	"helm.sh/helm/v3/pkg/action"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	k8smetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	helmChartReleaseHealthStatus = k8smetrics.NewGaugeVec(
+		&k8smetrics.GaugeOpts{
+			Name: "helm_chart_release_health_status",
+			Help: "Health of the Helm release",
+		},
+		[]string{"releaseName", "chartName", "chartVersion"},
+	)
+)
+
+func init() {
+	legacyregistry.MustRegister(helmChartReleaseHealthStatus)
+}
+
+func HandleHelmChartReleaseHealthStatus() {
+	defer recoverMetricPanic()
+
+	if actionConfig, err := getActionConfig(); err != nil {
+		klog.Errorf("metric helm_chart_release_health_status unhandled: %v", err)
+		return
+	} else {
+		listAction := action.NewList(actionConfig)
+		releases, err := listAction.Run()
+		if err != nil {
+			klog.Errorf("metric helm_chart_release_health_status unhandled: %v", err)
+			return
+		}
+
+		for _, release := range releases {
+			if status := release.Info.Status.String(); status == "deployed" {
```

Comment:
So only deploy is healthy, what about the other states. I would say if it is unknown or failed then is 0 then is 1 for all other states

Reply:
Updated